### ISSUE TITLE
Pyright fails to flag illegal generic exception in except clause

### ISSUE THREAD

**Issue by Tridwoxi** [microsoft/pyright#11186]
## Description

The runtime raises a `TypeError` on line 9 in the following example, so Pyright should flag it. Yet Pyright does not find any problems.

The `except` section of a `try-except` block wants a class that inherits from `BaseException`. The `GenericError` object is a class. However, the `GenericError[int]` object is not a class: `isinstance(GenericError[int], type)` is false. Hence it cannot be caught.

Relatedly, on `isinstance(GenericError[int], type)`, Basedpyright says `Unnecessary isinstance call; "type[GenericError[int]]" is always an instance of "type" (basedpyright reportUnnecessaryIsInstance)` but the runtime evaluates it as false.

## Example

In the folder "Bug":

```py
# src/example.py

class GenericError[T](Exception):
    def __init__(self, value: T) -> None:
        super().__init__()
        self.value: T = value


try:
    raise GenericError[int](1)  # Line 8.
except GenericError[int] as ge: # Line 9.
    print(ge.value)
```

```toml
# pyproject.toml

[tool.pyright]  # This table is removed when running basedpyright.

pythonVersion = "3.14"
typeCheckingMode = "strict"

[tool.basedpyright]  # This table is removed when running pyright.

pythonVersion = "3.14"
typeCheckingMode = "all"
```

```txt
Bug % pyright
0 errors, 0 warnings, 0 informations
Bug % basedpyright
0 errors, 0 warnings, 0 notes
Bug % python3 src/example.py
Traceback (most recent call last):
  File "/Users/tridwoxi/Desktop/Bug/src/example.py", line 8, in <module>
    raise GenericError[int](1)
GenericError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/tridwoxi/Desktop/Bug/src/example.py", line 9, in <module>
    except GenericError[int] as ge:
        print(ge.value)
TypeError: catching classes that do not inherit from BaseException is not allowed
```

## Environment

```txt
Bug % sw_vers
ProductName:		macOS
ProductVersion:		15.6.1
BuildVersion:		24G90
Bug % which python3
/opt/homebrew/bin/python3
Bug % python3 --version
Python 3.14.0
Bug % which pyright
/opt/homebrew/bin/pyright
Bug % pyright --version
pyright 1.1.407
Bug % which basedpyright
/opt/homebrew/bin/basedpyright
Bug % basedpyright --version
basedpyright 1.36.1
based on pyright 1.36.1
```

**Comment by erictraut** [comment]
I don't consider this a bug. There are many runtime errors that pyright is not designed to detect.

What you're asking for here is a new check. I'll relabel it as a feature request and consider it in that light.

I don't think this particular condition represents a common-enough condition to justify adding this new check into pyright, so I'm going to reject the feature request at this time. I'm willing to reconsider in the future if there's sufficient interest from pyright users (in the form of upvotes). For now, I'm going to close it.
